<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QLabel" name="original_image_label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="filtered_image_label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>100</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QCheckBox" name="normal_checkbox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Normal noise</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="mean_label">
            <property name="text">
             <string>Mean</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="mean_spinbox">
            <property name="maximum">
             <double>9999.989999999999782</double>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="std_label">
            <property name="text">
             <string>Std</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="std_spinbox">
            <property name="maximum">
             <double>9999.989999999999782</double>
            </property>
            <property name="value">
             <double>100.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QCheckBox" name="uniform_checkbox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Uniform noise</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="low_label">
            <property name="text">
             <string>Low</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="low_spinbox">
            <property name="maximum">
             <double>9999.989999999999782</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="high_label">
            <property name="text">
             <string>High</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="high_spinbox">
            <property name="maximum">
             <double>9999.989999999999782</double>
            </property>
            <property name="value">
             <double>255.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>100</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QPushButton" name="add_noise_button">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::StrongFocus</enum>
          </property>
          <property name="text">
           <string>Add noise</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="remove_noise_button">
          <property name="text">
           <string>Remove noise</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFilters">
    <property name="title">
     <string>Filters</string>
    </property>
    <addaction name="actionGauss_Filter"/>
    <addaction name="separator"/>
    <addaction name="actionMedian_Filter"/>
    <addaction name="separator"/>
    <addaction name="actionFilter2d"/>
    <addaction name="separator"/>
    <addaction name="actionSobel"/>
    <addaction name="separator"/>
    <addaction name="actionLaplacian"/>
   </widget>
   <addaction name="menuFilters"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionGauss_Filter">
   <property name="text">
    <string>Gauss Filter</string>
   </property>
  </action>
  <action name="actionMedian_Filter">
   <property name="text">
    <string>Median Filter</string>
   </property>
  </action>
  <action name="actionFilter2d">
   <property name="text">
    <string>Filter2d</string>
   </property>
  </action>
  <action name="actionSobel">
   <property name="text">
    <string>Sobel</string>
   </property>
  </action>
  <action name="actionLaplacian">
   <property name="text">
    <string>Laplacian</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>actionGauss_Filter</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>show_gauss_filter_widget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionMedian_Filter</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>show_median_filter_widget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFilter2d</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>show_filter2d_widget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSobel</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>show_sobel_widget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLaplacian</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>show_laplace_widget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>show_gauss_filter_widget()</slot>
  <slot>show_median_filter_widget()</slot>
  <slot>show_filter2d_widget()</slot>
  <slot>show_sobel_widget()</slot>
  <slot>show_laplace_widget()</slot>
 </slots>
</ui>
